### PR TITLE
Database Setup and Seed Migration

### PR DESCRIPTION
**Pull request by incredible-phoenix246**
#### Description

This PR sets up the database schema, configures TypeORM, and adds initial seed data for testing. The following changes were made:

1. **Entities Setup**
   - Created `User`, `Profile`, `Product`, and `Organization` entities with proper relationships.
   
2. **Migration**
   - Added migration to drop existing schema and recreate it.
   - Created seed migration to insert initial data into the database.

3. **Endpoints**
   - Implemented endpoints to fetch user data, including their profile, products, and organizations.

#### Changes

1. **Entity Changes**
   - Updated `User.ts` to include `@JoinColumn` for `@OneToOne` relationship with `Profile`.
   - Ensured proper use of `@OneToMany` and `@ManyToMany` relationships with appropriate join columns.

2. **Endpoints**
   - Implemented `GET /api/v1/users` to fetch all users.
   - Implemented `GET /api/v1/users/:id` to fetch a user by ID with their profile, products, and organizations.

#### How to Test

1. **Clone the Repository**
   ```bash
   git clone (https://github.com/hngprojects/hng_boilerplate_expressjs)
   cd hng_boilerplate_expressjs
   ```

2. **Install Dependencies**
   ```bash
   yarn install
   ```

3. **Configure Environment Variables**
   Ensure your `.env` file has the correct database configuration:
   ```env
port=8000
auth-secret=
DB_USER=
DB_HOST=
DB_PASSWORD=
DB_NAME=
   ```

4. **Start the Server**
   ```bash
   yarn start:dev
   ```

5. **Test Endpoints**
   - **Get All Users**
     ```bash
     curl http://localhost:8000/api/v1/users
     ```

   - **Get User by ID**
     Replace `{user_id}` with a valid user ID from the seeded data.
     ```bash
     curl http://localhost:8000/api/v1/users/{user_id}
     ```

6. **Verify Response**
   Ensure the response contains the expected user data, including profile, products, and organizations.

#### Notes

- Ensure the PostgreSQL server is running and accessible with the provided credentials.
- The database schema and initial data will be recreated, so any existing data will be lost.

---

By following these steps, you can verify the database setup, seed data, and endpoints are working correctly.

### CODE REVIEW

File: src/data-source.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import config from "./config";
+
+export const AppDataSource = new DataSource({
+  type: "postgres",
+  host: config.DB_HOST,
+  port: 5432,
+  username: config.DB_USER,
+  password: config.DB_PASSWORD,
+  database: config.DB_NAME,
+  synchronize: true,
```
Comment (NdubuisiJr) [inline src/data-source.ts:12]: This is wrong. Set `sychronize:false`. We only use `Migrations` for updating schema.

**Review by NdubuisiJr — DISMISSED** [review]

File: src/data-source.ts (hunk 1 above)
Comment (incredible-phoenix246) [reply src/data-source.ts]: updated @NdubuisiJr

**Review by Idimmusix — COMMENTED** [review]
Please don't push .env
Remove it from git using `git rm --cached .env`

**Review by nedssoft — CHANGES_REQUESTED** [review]
The task's specified that you should use migrations, models, service pattern.

The controller shouldn't access data directly from the model. The service can access the model directly while the controller accesses the model through the service.

**Comment by incredible-phoenix246** [comment]
@nedssoft @NdubuisiJr  all changes has been made

**Review by Idimmusix — APPROVED** [review]

**Pull request merged**